File: BDD/deployment_role.erl
% Copyright 2013, Dell 
% 
% Licensed under the Apache License, Version 2.0 (the "License"); 
% you may not use this file except in compliance with the License. 
% You may obtain a copy of the License at 
% 
%  http://www.apache.org/licenses/LICENSE-2.0 
% 
% Unless required by applicable law or agreed to in writing, software 
% distributed under the License is distributed on an "AS IS" BASIS, 
% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
% See the License for the specific language governing permissions and 
% limitations under the License. 
% 
% 
-module(deployment_role).
-export([step/2, json/3, validate/1, inspector/0, g/1]).
-include("bdd.hrl").

% Commont Routine
% Provide Feature scoped strings to DRY the code
g(Item) ->
  case Item of
    path  -> "/api/v2/deployment_roles";
    atom  -> bdd_deployment_role1;
    name  -> "bdd_deployment_role";
    _     -> crowbar:g(Item)
  end.
  

% Common Routine
% Makes sure that the JSON conforms to expectations (only tests deltas)
validate(JSON) when is_record(JSON, obj) ->
  J = JSON#obj.data,
  R =[JSON#obj.type == "deployment_role",
      bdd_utils:is_a(J, length, 5),
      bdd_utils:is_a(J, dbid, role_id),
      bdd_utils:is_a(J, dbid, deployment_id),
      crowbar_rest:validate_core(J)],
  bdd_utils:assert(R).

% Common Routine
% Returns list of nodes in the system to check for bad housekeeping
inspector() -> 
  bdd_restrat:inspector(deployment_role).  % shared inspector works here, but may not always

% Common Routine
% Creates JSON used for POST/PUT requests
json(Name, Description, Order) ->
  json:output([{"name",Name},{"description", Description}, {"order", Order}]).

     
% Common Routines

step(_Global, {step_setup, _N, _}) -> 
  % create node(s) for tests
  JSON = json(g(name), g(description), 100),
  bdd_crud:create(g(path), JSON, g(atom));

step(_Global, {step_teardown, _N, _}) -> 
  % find the node from setup and remove it
  bdd_crud:delete(g(atom)).

File: core/clients/python/api_bindings/cb2_api/objects/deployment_role.py
# Copyright 2014, Dell
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from cb2_api.endpoint import EndPoint
from apiobject import ApiObject

class Deployment_RoleEP(EndPoint):    
    '''
    https://github.com/digitalrebar/core/blob/master/doc/devguide/api/deployment_role.md
    '''
    __endpoint   = "/api/v2/deployment_roles"      
    __apiObjectType = "Deployment_Role"
    
    def __init__(self, session):
        self.session = session
        self.endpoint = Deployment_RoleEP.__endpoint
        super(Deployment_RoleEP, self).__init__(session)
        
class Deployment_Role(ApiObject):
    '''
    Deployment role object
    
    ''' 
    
    def __init__(self, json={}):
        self.wall = None
        self.created_at = None
        self.updated_at = None
        self.role_id = None
        self.deployment_id = None
        self.data = None
        self.id = None
        self.__dict__ = json
        super(Deployment_Role, self).__init__()
       
   
class Enums_Deployment():
    '''
    TODO
    '''
 
    
        

        
